docs/rfcs/1013-parsec-account.md

```diff
@@ -0,0 +1,376 @@
+
+#### Uploading a new device
+
+```yaml
+cmd: protected_device_upload
+req:
+  device: UploadedDevice
+reps:
+  ok: {}
+  # The server compare the human handle of the device to the email of parsec-account
+  invalid_human_handle: {}
+```
+
+```yaml
+label: UploadedDevice
+type: uploaded_device
+other_fields:
+  organization_id: OrganizationID
+  device_id: DeviceID
+  encrypted_device: EncryptedDevice
+```
+
+#### List devices
+
+To list the device linked to the authentication method's vault:
+
+```yaml
+cmd: protected_device_list
+req: {}
+reps:
+  ok:
+    devices: Vec<DeviceInfo>
+```
+
+`DeviceInfo` is an aggregation of [`ProtectedDevice`] and additional information retrieved from the metadata server.
+
+```yaml
+label: DeviceInfo
+type: device_info
+other_fields:
+  organization_id: OrganizationID
+  device_id: DeviceID
+  user_id: UserID
+  HumanHandle: HumanHandle
+  created_on: Timestamp
+  revoked_on: Option<Timestamp>
+  encrypted_device: EncryptedDevice
+```
+
+#### Recovery device list
+
+To list all protected device from all vault:
+
+```yaml
+cmd: protected_device_recovery_list
+req: {}
+reps:
+  ok:
+    vaults: Vec<VaultInfo>
+    devices: Vec<DeviceInfo>
+custom_types:
+  VaultInfo:
+    id: VaultID
+    created_on: Timestamp
+    created_by: Creator
+    auth_methods: Vec<AuthMethod>
+  AuthMethod:
+    algorithm: KeyAlgorithm
+    created_on: Timestamp
+    created_by: Creator
+```
+
+> [!CAUTION]
+> The returned list does not include device from the current vault (since we already have access to them and do not need to recover them)
+
+#### Updating the password
+
+```yaml
+cmd: auth_method_password_update
+
+```
+
+#### Rotate the vault key
+
+```yaml
+cmd: vault_key_rotation
+req:
+  devices: Vec<Device>
+  # A newly generated vault key encrypted by authentication secret key.
+  encrypted_vault_key: EncryptedSymKey
+reps:
+  ok: {}
+```
+
+Upon receiving this request, the server creates a new [`Vault`] and new [`ProtectedDevice`].
+
+> [!IMPORTANT]
+> Doing a vault rotation mean the other authentication methods would not have access to the new `vault_key`.
+> Meaning they would not access the devices (encrypted by the new `vault_key`).
+
+#### Account recovery
+
+```yaml
+cmd: account_recovery_start
+req:
+  email: EmailStr
+reps:
+  ok: {}
+```
+
+Similar to `account_send_email_validation_token`, the server will send an email to recover the account.
+If the email does not exist the server will still respond with OK status for the same reason as `account_send_email_validation_token`.
+
+```yaml
+cmd: account_recovery_finish
+req:
+  recovery_token: RecoveryToken
+  auth_method_key_paramsS: KeyAlgorithm
+  encrypted_vault_key: EncryptedSymKey
+reps:
+  ok: {}
+  invalid_recovery_token: {}
+```
+
+The server will create a new [`Vault`] & [`AuthMethod`]
+
+#### Account deletion
+
+```yaml
+cmd: account_delete
+req: {}
+reps:
+  ok: {}
+```
+
+The server will send an email with a link to confirm the suppression of the account.
```

Comment:
Will the link use a route like the following ?
```yaml
cmd: account_delete_validate
req:
    delete_token: DeleteToken
reps:
    ok: {}
    invalid_token: {}
```

Reply:
I've imagined that we would use a standalone link, would it be better to reuse the same system for the account creation? (where the user would need to enter the token in a input field that `libparsec` does the final request)

Reply:
Good point. Taking into account the importance of the information stored AND that this action cannot be undone, it seems reasonable to protect the user against mistakes. 

So yes, we should probably use the send-token-to-confirm-deletion approach.

Reply:
So not a standalone link send in the email?

Reply:
Conclusion, we will use a token with an authenticated

Reply:
done